silq/pulses/pulse_sequences.py

```diff
@@ -596,6 +603,10 @@ def add_NMR_pulses(self, pulse_sequence=None):
                     int_pulse, = self.add(intermediate_pulse)
                     t_connect(int_pulse['t_start'])
                     t_connect = partial(int_pulse['t_stop'].connect, offset=0)
+            elif self.NMR['post_pulse'] is not None:
```

Comment:
Do you only want the post-pulse to be added when there aren't any `NMR_pulses`?
I thought `post_pulse` would always be appended to the end.

Reply:
No, (self.NMR['post_pulse'] is not None) - this line of code you are referring to is to insure that post_pulse is added at the end of all NMR pulses only if it exists.

Reply:
Ah I see, that makes sense because it's in the NMR settings dictionary, my mistake.